pydarn/io/supdarn_io.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2020 SuperDARN Canada, University of Saskatchewan
+# Author: Marina Schmidt
+
+import bz2
+import pydarnio
+import os
+
+
+class SuperDARNRead(pydarnio.SDarnRead):
+    """
+    A class that reads select SuperDARN files for pyDARN plotting
+
+    Methods
+    -------
+    read_dmap : reads superDARN DMap formats
+    read_borealis: not implemented yet
+    """
+    def __init__(self):
+        pass
+
+    def read_dmap(self, filename: str):
+        """
+        Reads select SuperDARN DMap files for pyDARN plotting:
+            fitacf
+            rawacf
+            iqdat
+        When more plotting methods are introduced other files
+        will be included.
+
+        Parameters
+        ----------
+            filename: string
+                name of the file you are going to read
+
+        Returns
+        ------
+            data: List[dict]
+                data records from the file
+        """
+        # check if the file  is compressed with
+        # bz2
+        if 'bz2' in filename:
+            with bz2.open(filename) as fp:
+                dmap_stream = fp.read()
+
+            super().__init__(dmap_stream, stream=True)
+        else:
+            super().__init__(filename)
+
+        try:
+            # Check which file type it is
+            if 'rawacf' in filename:
+                data = self.read_rawacf()
+            elif 'fitacf' in filename:
+                data = self.read_fitacf()
+            elif 'iqdat' in filename:
+                self.read_iqdat()
+            # if not noticeable then just read the file
+            else:
+                data = self.read_records()
+        except Exception as err:
+            # sometimes files might fail due to issues with
+            # the specific reading methods. This is not a pyDARN
+            # issue so we will just read the dmap file if that fails
+            # then make an issue on pyDARNio
+            print(err)
+            print("..... Will try to read DMap file with read_dmap")
+            print(" IF THIS FAILS please make an issue on pyDARNio, not "
+                  "pyDARN's issue")
+            data = self.read_records
+        return data
+
+    def read_borealis(self, filename: str, slice_id: int = None):
+        """
+        Reads RAWACF or BFIQ borealis files and converts them to
+        pyDARN data format for plotting.
```

Comment:
Can we say this? I was hoping one day we could do some sort of mapping for the original Borealis fields so that they could be plotted by pydarn without the conversion, and I think saying SDARN data is more specific.

```suggestion
        an SDARN data format dictionary for plotting.
```